floors/rule.go

```diff
@@ -0,0 +1,504 @@
+			value = request.App.Publisher.Domain
+		}
+	}
+	return value
+}
+
+func getSiteDomain(request *openrtb2.BidRequest) string {
+	var value string
+	if request.Site != nil {
+		value = request.Site.Domain
+	} else {
+		value = request.App.Domain
+	}
+	return value
+}
+
+func getPublisherDomain(request *openrtb2.BidRequest) string {
+	value := catchAll
+	if request.Site != nil && request.Site.Publisher != nil && len(request.Site.Publisher.Domain) > 0 {
+		value = request.Site.Publisher.Domain
+	} else if request.App != nil && request.App.Publisher != nil && len(request.App.Publisher.Domain) > 0 {
+		value = request.App.Publisher.Domain
+	}
+	return value
+}
+
+func getBundle(request *openrtb2.BidRequest) string {
+	value := catchAll
+	if request.App != nil && len(request.App.Bundle) > 0 {
+		value = request.App.Bundle
+	}
+	return value
+}
+
+func getgptslot(imp openrtb2.Imp) string {
+	value := catchAll
+	adsname, err := jsonparser.GetString(imp.Ext, "data", "adserver", "name")
+	if err == nil && adsname == "gam" {
+		gptSlot, _ := jsonparser.GetString(imp.Ext, "data", "adserver", "adslot")
+		if gptSlot != "" {
+			value = gptSlot
+		}
+	} else {
+		value = getpbadslot(imp)
+	}
+	return value
+}
+
+func extractChanelNameFromBidRequestExt(bidRequest *openrtb2.BidRequest) string {
+	requestExt := &openrtb_ext.ExtRequest{}
+	if bidRequest == nil {
+		return catchAll
+	}
+
+	if len(bidRequest.Ext) > 0 {
+		err := json.Unmarshal(bidRequest.Ext, &requestExt)
+		if err != nil {
+			return catchAll
+		}
+	}
+
+	if requestExt.Prebid.Channel != nil {
+		return requestExt.Prebid.Channel.Name
+	}
+	return catchAll
+}
+
+func getpbadslot(imp openrtb2.Imp) string {
+	value := catchAll
+	pbAdSlot, err := jsonparser.GetString(imp.Ext, "data", "pbadslot")
+	if err == nil {
+		value = pbAdSlot
+	}
+	return value
+}
+
+func getAdUnitCode(imp openrtb2.Imp) string {
+	adUnitCode := catchAll
+	gpId, err := jsonparser.GetString(imp.Ext, "gpid")
+	if err == nil && gpId != "" {
+		return gpId
+	}
+
+	if imp.TagID != "" {
+		return imp.TagID
+	}
+
+	pbAdSlot, err := jsonparser.GetString(imp.Ext, "data", "pbadslot")
+	if err == nil && pbAdSlot != "" {
+		return pbAdSlot
+	}
+
+	storedrequestID, err := jsonparser.GetString(imp.Ext, "prebid", "storedrequest", "id")
+	if err == nil && storedrequestID != "" {
+		return storedrequestID
+	}
+	return adUnitCode
+}
+
+func isMobileDevice(userAgent string) bool {
+	isMobile, err := regexp.MatchString("Phone|iPhone|[Mobile].*Android|[Android].*Mobile", userAgent)
+	if err != nil {
+		return false
+	}
+	return isMobile
+}
+
+func isTabletDevice(userAgent string) bool {
+
+	isTablet, err := regexp.MatchString("tablet|iPad|Android|[Windows NT].*touch|[touch].*Windows NT", userAgent)
+	if err != nil {
+		return false
+	}
+	return isTablet
+}
+
+func prepareRuleCombinations(keys []string, numSchemaFields int, delimiter string) []string {
+	var subset []string
+	var comb []int
+	var desiredkeys [][]string
```

Comment:
Camel-Case comment: `desiredkeys` --> `desiredKeys`

Reply:
renamed to `desiredKeys`